lucille-core/src/main/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnector.java

```diff
@@ -62,123 +68,117 @@ public DatabaseConnector(Config config) {
   }
 
   // create a jdbc connection
-  private void createConnection() throws ClassNotFoundException, SQLException {
+  private Connection createConnection() throws ClassNotFoundException, SQLException {
+    Connection connection = null;
     try {
       Class.forName(driver);
     } catch (ClassNotFoundException e) {
-      // TODO better error handling/logging
-      e.printStackTrace();
+      log.error("Driver not found {} check classpath to make sure the jdbc driver jar file is there.", driver);
       setState(ConnectorState.ERROR);
       throw(e);
     }
     try {
       connection = DriverManager.getConnection(connectionString, jdbcUser, jdbcPassword);
     } catch (SQLException e) {
-      // TODO better logging/error handling.
-      e.printStackTrace();
+      log.error("Unable to connect to database {} user:{}", connectionString, jdbcUser);
       setState(ConnectorState.ERROR);
       throw(e);
     }
+    connections.add(connection);
+    return connection;
   }
 
-  private void setState(ConnectorState error) {
-    // TODO: update the State here.
-    // System.err.println("Implement Me: ");
+  private void setState(ConnectorState newState) {
+    this.state = newState;
   }
 
   @Override
   public void execute(Publisher publisher) throws ConnectorException {
-    
-    try {
-    setState(ConnectorState.RUNNING);
-    // connect to the database.
-    createConnection();
-    // run the pre-sql (if specified)
-    runSql(preSql);
 
-    ResultSet rs = null;
     try {
+      setState(ConnectorState.RUNNING);
+      // connect to the database.
+      Connection connection = createConnection();
+      // run the pre-sql (if specified)
+      runSql(connection, preSql);
+      // TODO: make sure we cleanup result set/statement/connections properly.

```

Comment:
Is there somewhere we know or suspect we're not cleaning up connections?

Reply:
We should go through all TODOs in the codebase, but since that's not part of a regular process yet, how about making a ticket for this -- or fixing it with this PR?

Reply:
Created a ticket LC-316 to address connection cleanup so that the merging of this PR is not otherwise blocked.